packages/transformers/typescript-tsc/src/TSCTransformer.js

```diff
@@ -0,0 +1,82 @@
+// @flow strict-local
+
+import {Transformer} from '@parcel/plugin';
+import localRequire from '@parcel/local-require';
+
+type TypescriptCompilerOptions = {
+  module?: mixed,
+  jsx?: mixed,
+  noEmit?: boolean,
+  sourceMap?: boolean
+};
+
+type TypescriptTranspilerOptions = {
+  compilerOptions: TypescriptCompilerOptions,
+  fileName: string
+};
+
+export default new Transformer({
+  async getConfig({asset}) {
+    return asset.getConfig(['tsconfig.json']);
```

Comment:
This should use the compiler's `findConfigFile` api rather than our own [0]

We should also make sure it resolves `extends` fields present in configs -- the version here certainly won't.

[0] https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API

Reply:
cc @padmaia

Reply:
Looks like Fathy's plugin resolves the initial config and passes it to the compiler options. Is there something keeping you from doing something similar right now? Also I wonder if the compiler tells us which files it extended. Probably not as most tools don't, but I wonder if it would be difficult to get that feature in. Otherwise we wouldn't be able to rebuild in watch mode if one of the extended files changed and would have to rerun loading the tsconfig for every file whose tsconfig had an extends on start up 😕 (Same as babel, @wbinnssmith you already know most of this, just commenting for posterity)